private/pkg/thread/thread_test.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package thread
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/atomic"
+)
+
+func TestParallelizeWithImmediateCancellation(t *testing.T) {
+	// The bulk of the code relies on subtle timing that's difficult to
```

Comment:
Needs a t.Parallel(), im surprised lint didnt catch this

Reply:
Sure, can do.

Reply:
I'm tempted to make one of those Xzibit memes about parallel testing for parallel, but even reading the template confuses me.